runelite-client/src/main/java/net/runelite/client/plugins/kingdomofmiscellania/KingdomPlugin.java

```diff
@@ -122,20 +207,92 @@ private void removeKingdomInfobox()
 		}
 	}
 
+	private int estimateCoffer(long lastChangedEpoch, int lastCoffer)
```

Comment:
You already have the last changed as an `Instant` so there's no reason to convert to epoch second and then back like you are doing here and in `estimateFavor`.